compiler/main.nim

```diff
@@ -363,8 +363,11 @@ proc mainCommand*(graph: ModuleGraph) =
                 elif isDefined(conf, "release"): "Release"
                 else: "Debug"
     let sec = formatFloat(epochTime() - conf.lastCmdTime, ffDecimal, 3)
-    let project = if optListFullPaths in conf.globalOptions: $conf.projectFull else: $conf.projectName
-    let output = if optListFullPaths in conf.globalOptions: $conf.getOutFileFull else: $conf.outFile
+    template postprocess(path): untyped =
+      if optListFullPaths in conf.globalOptions: $path
+      else: path.`$`.lastPathPart
```

Comment:
I don't understand this.

Reply:
you had commented here https://github.com/nim-lang/Nim/pull/13043#issuecomment-570941009
> I see the value in it but omg it's verbose

and this allows shortening things when `--listfullpaths:off` (as was done here https://github.com/nim-lang/Nim/pull/13043)

that's the simplest code I can think of that does what we want

Reply:
``lastPathPart`` is never what I want as my filenames do not in end in slashes anyway. The old code

```nim

let project = if optListFullPaths in conf.globalOptions: $conf.projectFull else: $conf.projectName
let output = if optListFullPaths in conf.globalOptions: $conf.getOutFileFull else: $conf.outFile

```

was perfectly clear code. Clearer than your replacement which I still don't understand.

Reply:
old code was incorrect because of the logic inside `absOutFile` which sometimes adds `.out` suffix and wasn't reflected here; I've now fixed it differently without using lastPathPart and it should work in all cases